anesthetic/_matplotlib/boxplot.py

```diff
@@ -0,0 +1,129 @@
+            y = np.atleast_2d(y)
+            bp = ax.bxp([_bxpstat(yi, weights, whis) for yi in y], **kwds)
+
+        if return_type == "dict":
+            return bp, bp
+        elif return_type == "both":
+            return cls.BP(ax=ax, lines=bp), bp
+        else:
+            return ax, bp
+
+    def _make_plot(self):
+        if isinstance(self.data, _WeightedObject):
+            self.kwds['weights'] = self.data.weights
+        return super()._make_plot()
+
+
+def boxplot(data, *args, **kwds):
+    # noqa: disable=D103
+    if isinstance(data, _WeightedObject):
+        kwds['weights'] = data.weights
+
+    def create_plot_group():
+        fontsize = None  # pragma: no cover
+        kwds = None  # pragma: no cover
+        maybe_color_bp = None  # pragma: no cover
+        return_type = None  # pragma: no cover
+        rot = None  # pragma: no cover
+
+        def plot_group(keys, values, ax, **kwargs):  # pragma: no cover
+            # GH 45465: xlabel/ylabel need to be popped out before plotting
+            xlabel = kwds.pop("xlabel", None)
+            ylabel = kwds.pop("ylabel", None)
+            if xlabel:
+                ax.set_xlabel(pprint_thing(xlabel))
+            if ylabel:
+                ax.set_ylabel(pprint_thing(ylabel))
+
+            weights = kwds.pop("weights", None)
+            keys = [pprint_thing(x) for x in keys]
+            if weights is None:
+                values = [np.asarray(remove_na_arraylike(v), dtype=object)
+                          for v in values]
+                bp = ax.boxplot(values, **kwds)
+            else:
+                whis = kwds.pop("whis", 1.5)
+                kwds['showfliers'] = False
+                from anesthetic._matplotlib.boxplot import _bxpstat
+                bp = ax.bxp([_bxpstat(v, weights, whis) for v in values],
+                            **kwds)
+
+            if fontsize is not None:
+                ax.tick_params(axis="both", labelsize=fontsize)
+
+            # GH 45465: x/y are flipped when "vert" changes
+            is_vertical = kwds.get("vert", True)
+            ticks = ax.get_xticks() if is_vertical else ax.get_yticks()
+            if len(ticks) != len(keys):
+                i, remainder = divmod(len(ticks), len(keys))
+                assert remainder == 0, remainder
+                keys *= i
+            if is_vertical:
+                ax.set_xticklabels(keys, rotation=rot)
+            else:
+                ax.set_yticklabels(keys, rotation=rot)
+            maybe_color_bp(bp, **kwds)
+
+            # Return axes in multiplot case, maybe revisit later # 985
+            if return_type == "dict":
+                return bp
+            elif return_type == "both":
+                return BoxPlot.BP(ax=ax, lines=bp)
+            else:
+                return ax
+
+    boxplot = replace_inner_function(pandas.plotting._matplotlib.boxplot,
+                                     create_plot_group.__code__.co_consts[1])
```

Comment:
Use the rest of the original `pandas.plotting._matplotlib.boxplot` function, but overwrite the inner `plot_group` function which calls the unweighted `ax.boxplot`